api/api.go

```diff
@@ -132,3 +136,27 @@ func handleBodyUnmarshalError(ctx context.Context, err error) *models.ErrorRespo
 		nil,
 	)
 }
+
+func initialiseRoleGroups(ctx context.Context, cognitoClient cognito.Client, userPoolId string) error {
+	adminGroup := models.NewAdminRoleGroup()
+	adminGroupCreateInput := adminGroup.BuildCreateGroupRequest(userPoolId)
+	_, err := cognitoClient.CreateGroup(adminGroupCreateInput)
+	if err != nil {
+		cognitoErr := models.NewCognitoError(ctx, err, "CreateGroup request for admin group from API start up")
+		if cognitoErr.Code != models.GroupExistsError {
+			return cognitoErr
+		}
```

Comment:
Just making sure I follow this logic: If there is an error we only return it if its a Group exists error?

Reply:
No we return any errors that aren't `GroupExistsError`.
`GroupExistsError`s are swallowed.

The purpose of this function is to make sure the role groups are setup in Cognito, creating them if they aren't there.
If Cognito returns a success or a `GroupExistsError` then the group is in place and the service is ok to run. Any other error needs to be returned and stop the service starting.